packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -81,7 +83,148 @@ class AnimatedIcon extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    // TODO(amirh): implement this.
-    return new Container();
+    // TODO(amirh): implement semantics, text direction, scaling.
+    final _AnimatedIconData iconData = icon;
+    return new CustomPaint(
+      size: iconData.size,
+      painter: new _AnimatedIconPainter(
+        iconData.paths,
+        progress.value,
+        color,
+        () => new ui.Path(),
```

Comment:
i think Path should be visible from the material library here, you might not need to use `ui.Path`.

Reply:
(or maybe the painting library)

Reply:
I actually prefer it to explicitly be ui.Path as Path is a bit overloaded term in this area.

Let me know if you still prefer Path